msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java

```diff
@@ -65,6 +65,8 @@ protected void onCreate(final Bundle savedInstanceState) {
         if (savedInstanceState != null) {
             Logger.verbose(TAG, null, "AuthenticationActivity is re-created after killed by the os.");
             mRestarted = true;
+            mTelemetryRequestId = savedInstanceState.getString(Constants.TELEMETRY_REQUEST_ID);
+            mUiEventBuilder = new UiEvent.Builder();
```

Comment:
What's the advantage? Making that change requires inverting the order of [these two checks](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java#L64-L75) and I _still_ need to conditionally choose whether I am going to hydrate `mRequestId` from `savedInstanceState` or from `getIntent()`. The consequences of inverting these checks are also not immediately apparent to me: they both control early `return/sendError()` conditions: I'm inclined to leave as-is